### ISSUE TITLE
Error when predictions have extra dimensions other than classes

### ISSUE THREAD

**Issue by confifu** [HanxunHuangLemonBear/SCELoss-Reproduce#4]
torch.nn.CrossEntropyLoss takes prediction of shape (minibatch, Classes, d1, d2 , ...). This line leads to error if the d1, d2 dimensions exist.
https://github.com/HanxunH/SCELoss-Reproduce/blob/c81fb33638087246da70fc157ea48ce43542d3fc/loss.py#L23
The predictions are of shape (minibatch, classes, d1, d2, ...) but the `label_one_hot` variable is of shape (minibatch, d1, d2, ..., classes). I think the `label_one_hot` tensor should be transposed to make the class dimension the second dimension.

**Comment by HanxunH** [comment]
Hi,
Thanks for the suggestion! Feel free to open a PR.
